Core/src/RequestWrapperTrait.php

```diff
@@ -178,11 +178,17 @@ public function getCredentialsFetcher()
             }
         }
 
-        return new FetchAuthTokenCache(
-            $fetcher,
-            $this->authCacheOptions,
-            $this->authCache
-        );
+        if (\is_a($fetcher, FetchAuthTokenCache::class)) {
```

Comment:
`instanceof` is preferred: 

```suggestion
        if ($fetcher instanceof FetchAuthTokenCache::class) {
```